templates/server-deployment.yaml

```diff
@@ -117,6 +117,7 @@ spec:
             - name: TEMPORAL_VERSION_CHECK_DISABLED
               value: "1"
           {{- end }}
+          {{- include "additional-env" (default $.Values.server.additionalEnv $serviceValues.additionalEnv) | indent 10 }}
```

Comment:
why is `additional-env` needed? why not handle this the same way that `additionalVolumeMounts` is handled? If there is a reason why the `additional-env` approach is better could we replace all the other handling for lists of additional things?

Reply:
You are right. I pushed the rebase and added more cases in PR description.